README.md

```diff
@@ -1,41 +1,112 @@
+
+<ul>
+AIM Minimum Rank – Special Graphs Work Group. Zero forcing sets and the minimum rank of graphs. <i>Linear Algebra and its Applications</i>, 428(7):1628–1648 (2008). (<a href="https://doi.org/10.1016/j.laa.2007.10.009">https://doi.org/10.1016/j.laa.2007.10.009</a>)
+</ul>
+
+### Definition.
+
+Given a finite graph, choose some vertices to be “filled” and then apply this rule repeatedly until it cannot be applied further: If any filled vertex has exactly one unfilled neighbor, then that neighbor becomes filled.  (We say that the first vertex “forces” its neighbor.)  The smallest number of vertices that can be initially filled so that all vertices become filled eventually is the **zero forcing number** of the graph.
+
+### Method.
+
+The algorithm applied finds the least weight of a path in a directed, weighted “metagraph” in which each vertex represents a subset of vertices in the primal graph *G* (the one whose zero forcing number is desired) with the property that, when this subset is precisely the set of filled vertices, no vertex can force.  An arc of weight *w* is present from *X* to *Y* when it is possible to add *w* vertices to the initially filled set that produced *X* to obtain an initially filled set that produces *Y*.  (That is, it is possible to expand the size of the ultimately filled set from *X* to *Y* at the “cost” of filling *w* additional vertices at the beginning.)  Then one can show that the smallest total weight of a directed path from &empty; to *V(G)* in this metagraph is the zero forcing number of *G*.
+
+The advantage to the above formalism is that it allows the zero forcing number to be computed using any of the standard algorithms for finding a path of smallest weight in a weighted directed graph.  In particular, in this implementation we use Dijkstra's Algorithm.
+
+### Capabilities and limitations.
+
+The software is able to compute the zero forcing number very efficiently for most simple graphs.  Some graphs, such as stars, represent a weakness for the algorithm and may produce longer running times.  In addition, memory usage can be prodigious for very large graphs.  Steps are planned to address both of these limitations in the future.  For now, the algorithm is very quick for most graphs.  For example, for the Paley graph on 101 vertices, the zero forcing number can be computed in just a few seconds.
 
 ---
 
-## Install from PyPI
-```bash
-sage -pip install -i https://test.pypi.org/simple/ zeroforcing
+
+### How to install:
+
+#### Option 1. Install through PyPI
+
+Type the line below into any cell in a SageMath Jupyter notebook:
+
+```
+%pip install --index-url https://test.pypi.org/simple/ zeroforcing==0.1.0
```

Comment:
We're going to remove the `--index-url [url]` portion when we upload the first version to the Prod PyPI repo, but I think we should omit the version number. Keeping it in here means we'd have to change this line every time we change the version. If you just install `zeroforcing` (will be called `zftools`), it will grab the latest version.

Reply:
Right.  Agree this obviously needs to change.  What should the new line look like then?  Maybe:
```
%pip install zftools
```
Can it really be as simple as just that?

Reply:
Yep, that's how it'll look! Cool, right?

Reply:
Actually, we might need to stick a `--no-build-isolation` flag in there, since pip builds packages in a clean virtual environment by default. (Similar to what we do in the [Dockerfile](https://github.com/alexhutman/ZeroForcingNumber/blob/78c76cd615c640cc18ee4278885091efee01a130/Dockerfile#L7), except the users won't be invoking the `build` module manually, they'll use the [equivalent flag](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-no-build-isolation) in pip.) This allows the build process to use Sage's Cython version as well as Sage env vars, and Sage's `build_ext` subclass (used [here](https://github.com/alexhutman/ZeroForcingNumber/blob/78c76cd615c640cc18ee4278885091efee01a130/build_helper.py#L8) in our build process), where it (presumably) includes Sage's headers and stuff during Cython builds.

Reply:
I tested via the Sage shell and `--no-build-isolation` was not needed. You can verify this by removing everything except the `FROM` line in the Dockerfile and simply running `pip install zftools`. I'm actually not quite sure why it works though, now that I think about it... I wonder if Sage has some custom pip override where they always force that flag?